lms/djangoapps/course_goals/views.py

```diff
@@ -0,0 +1,79 @@
+"""
+Course Goals Views - includes REST API
+"""
+from django.contrib.auth import get_user_model
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from eventtracking import tracker
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.lib.api.permissions import IsStaffOrOwner
+from rest_framework import permissions, serializers, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from .models import CourseGoal
+
+
+User = get_user_model()
+
+
+class CourseGoalSerializer(serializers.ModelSerializer):
+    """
+    Serializes CourseGoal models.
+    """
+    user = serializers.SlugRelatedField(slug_field='username', queryset=User.objects.all())
+
+    class Meta:
+        model = CourseGoal
+        fields = ('user', 'course_key', 'goal_key')
+
+    def validate_course_key(self, value):
+        """
+        Ensure that the course_key is valid.
+        """
+        course_key = CourseKey.from_string(value)
```

Comment:
Note: We don't verify that the user can access the course anywhere like @clintonb did in his [cms api](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/api/v1/views/course_runs.py#L37):
```
course_run = get_course_and_check_access(course_run_key, self.request.user)
```
Are we ok without this if it isn't a big deal if someone creates bad data?